src/utils/deepAssign.js

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isObject from "./isObject";
+
+const deepAssignObject = (target, source) => {
+  Object.keys(source).forEach(key => {
+    if (isObject(target[key]) && isObject(source[key])) {
+      deepAssignObject(target[key], source[key]);
+
+      return;
+    }
+
+    target[key] = source[key];
+  });
+};
+
+/**
+ * Recursively copy the values of all enumerable own properties from a source item to a target item if the both items are objects
+ * @private
```

Comment:
It's interesting that you marked this as private. Is that because you wouldn't want it to show up in our JSDoc if we ever generated it? It's an interesting thing to think about...whether our produced JSDoc would be intended for Alloy customers or Alloy developers. I would probably think we would ever only produce JSDoc for other developers, if ever. 🤷‍♂

Reply:
My bad, I just copied it from another place. Right I am not sure we discussed about JSDoc and the way we want to is it. For now I just removed `private` since it doesn't make any sense.